src/modules/launcher/Wox/ViewModel/ResultViewModel.cs

```diff
@@ -22,7 +22,7 @@ public enum ActivationType
             Hover
         };
 
-        public List<ContextMenuItemViewModel> ContextMenuItems { get; set; }
+        public ObservableCollection<ContextMenuItemViewModel> ContextMenuItems { get; set; }

```

Comment:
You still need to raise OnPropertyChanged when you change reference to ContextMenuItem. It will only help you if you do not recreate this property. So you would set it here:
`public ObservableCollection<ContextMenuItemViewModel> ContextMenuItems { get; set; } = new ObservableCollection<ContextMenuItemViewModel>();
`
 And everywhere where you use it you just call:
ContextMenuItems.Clear();
...
and then you add items directly to this property - `ContextMenuItems.Add(item)`; not by creating a new list and setting that to this property.

If you don't want to change all of that, just revert it to your previous version, that will work OK

Reply:
Just curious, all the documentation seems to say that the class/viewmodel must implement `INotifyPropertyChanged`, which `observableCollection` does. Why does initializing it to a `new` observableCollection cause a memory leak and how would that be different from the initial  `ObservableCollection<ContextMenuItemViewModel> ContextMenuItems { get; set; } = new ObservableCollection<ContextMenuItemViewModel>();`? I'm trying to understand why this is ok but not doing the same elsewhere in code. I was thinking that it would still be bound using `PropertyChanged` and not `ValueChanged`, therefore would not have a strong reference.

Reply:
Changes looks good to me now. In this case, the problem was that you are replacing your property with a new reference. Therefore View does know, that something has been changed because it was "tracking" a different reference before. ObservableCollection is providing notifications when something inside changes (added/removed items) , therefore View is notified about the change and can react on that. 
In this cause, calling new on observable collection would not cause memory leak, but view would not react on changes anymore because you just replaced it with a whole new object.

Reply:
Thank you for the clarification. It makes sense now, I was thinking only along the lines of the memory leak.